Assets/Scenes/SubBoom.cs

```diff
@@ -231,32 +296,52 @@ void Update()
         }
 
         // Update submarine positions
-        foreach (var sub in submarines) {
-            sub.Update(Time.deltaTime, bubbles);
+        foreach (var sub in submarines)
+        {
+            sub.UpdateMovement(Time.deltaTime, bubbles);
+
+            torpedoTimeUntilLaunch += Time.deltaTime;
+            if (torpedoTimeUntilLaunch == torpedoSpawnTime)
```

Comment:
I think it's safer to write `torpedoTimeUntilLaunch >= torpedoSpawnTime` since it's possible for the float addition to jump over the guard value and skip over this conditional code (i.e. launch time could be something like 0.99999 then 1.0000001 and spawn time could be 1 and the if statement would be false). The greater than or equal to will make sure to trigger whenever the launch time has elapsed.